src/Mealz/UserBundle/Provider/OAuthUserProvider.php

```diff
@@ -68,49 +61,52 @@ public function loadUserByUsername($username)
     /**
      * Loads an user by identifier or create it.
      *
-     * @param Object $userInformation The user information
+     * @param string $username
+     * @param array $userInformation The user information
      *
      * @return OAuthUser|boolean
      */
-    public function loadUserByIdOrCreate($userInformation)
+    public function loadUserByIdOrCreate($username, $userInformation)
     {
-        // First Check if array Informations are given
-        if (gettype($userInformation) === 'object' &&
-            property_exists($userInformation, 'preferred_username') === true &&
-            property_exists($userInformation, 'family_name') === true &&
-            property_exists($userInformation, 'given_name') === true
+        // First Check if all informations are given
+        if (empty($username) === false &&
+            gettype($userInformation) === 'array' &&
+            array_key_exists('family_name', $userInformation) === true &&
+            array_key_exists('given_name', $userInformation) === true
         ) {
             $profile = $this->doctrineRegistry->getManager()->find(
                 'MealzUserBundle:Profile',
-                $userInformation->preferred_username
+                $username
             );
         } else {
             return false;
         }
 
-        //if Userprofile is null, create User
+        // When Userprofile is null, create User
         if ($profile === null) {
             $profile = $this->createProfile(
-                $userInformation->preferred_username,
-                str_replace(' ', '', explode(',', $userInformation->given_name)[1]),
-                $userInformation->family_name
+                $username,
+                str_replace(' ', '', explode(',', $userInformation['given_name'])[1]),
```

Comment:
Wäre es möglich, diese Anweisung mit einem Kommentar zu versehen oder für eine bessere Verständlichkeit in eine Funktion auszulagern?

Reply:
done